lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1307,6 +1308,63 @@ def upload_exec_summary_report(_xmodule_instance_args, _entry_id, course_id, _ta
     return task_progress.update_task_state(extra_meta=current_step)
 
 
+def upload_course_survey_report(_xmodule_instance_args, _entry_id, course_id, _task_input, action_name):  # pylint: disable=invalid-name
+    """
+    For a given `course_id`, generate a html report containing the survey results for a course.
+    """
+    start_time = time()
+    start_date = datetime.now(UTC)
+    num_reports = 1
+    task_progress = TaskProgress(action_name, num_reports, start_time)
+
+    current_step = {'step': 'Gathering course survey report information'}
+    task_progress.update_task_state(extra_meta=current_step)
+
+    distinct_survey_fields_queryset = SurveyAnswer.objects.values('field_name').distinct()
```

Comment:
we need a filter on course_id here, we just want distinct names for the course. Different courses use different surveys and each will have a different set of distinct field_names